janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/BufferUtil.java

```diff
@@ -57,6 +65,50 @@ public static StaticBuffer getIntBuffer(int[] ids) {
         return StaticArrayBuffer.of(arr);
     }
 
+    /**
+     * We always add a byte marker at the beginning to indicate it's a string buffer. Then we
+     * write the string uncompressed. Finally, if the total buffer size is 8, we add a dummy byte
+     * at the end so that upon read time, JanusGraph knows the buffer does not store a long value.
+     * See {@link IDManager#getKeyID(StaticBuffer)} for more details.
+     *
+     * Note that we apply STOP_MASK to the last character. We don't have to do it because a static buffer
+     * has a fixed length, and thus upon read time, we know where to stop reading the string. This is more
+     * to keep it consistent with {@link StringEncoding#writeAsciiString(byte[], int, String)} where we use
+     * STOP_MASK to mark the end of the string. An additional benefit of doing so is to enable corruption check
+     * in the future. Note: this approach has no overhead.
+     *
+     * @param s
+     * @return
+     */
+    public static StaticBuffer getStringBuffer(String s) {
+        if (StringUtils.isEmpty(s) || !StringUtils.isAsciiPrintable(s)) {
+            throw new IllegalArgumentException("value must be non-empty printable ASCII string but received: " + s);
+        }
+        ByteBuffer buffer;
+        if (s.length() + byteSize == longSize) {
+            buffer = ByteBuffer.allocate(longSize + byteSize);
+        } else {
+            buffer = ByteBuffer.allocate(s.length() + byteSize);
+        }
+        buffer.put(STRING_ID_MARKER);
```

Comment:
Ping me when this comment is implemented: https://github.com/JanusGraph/janusgraph/pull/3418/files#r1173201627  (more discussion about what each of those 8 bits will mean is here: https://github.com/JanusGraph/janusgraph/issues/3732#issue-1678683473 )

If you added that logic somewhere, but I missed it, please, ping me and I will take a look at it.

Reply:
I didn't add that logic yet. For ASCII string type, we don't need to rely on a value size detection bit (because the final non-trivial byte is masked (`if (i+1==s.length()) b |= STOP_MASK;`).

I agree that can be very useful for other non-fixed-length types, e.g. byte array. I just feel like before we actually implement that functionality, there's no need to make the contract fixed. In other words, right now I feel like a value size detection bit is a very good idea for a byte array type, but once we really implement support for byte array ID, we may come up with a different approach.

Reply:
That make sense. I think this bit will be necessary for byte[] only type. Thus, should be fine. 
I initially thought that we need this bit for String as well, but got it that we don’t need it. 
I think it would be nice to add 2 tests just in case where we decode two strings of size == 9 with one string being 7 in original size and another one being 8 in original size. 
This is just in case we decide to make some breaking changes and forget about this case.

Reply:
This is covered in `JanusGraphCustomIdTest.java#testMixedStringAndLongVertexId` but I'll add two additional tests to be safe.